src/backend/dx12/src/device.rs

```diff
@@ -1446,38 +1446,52 @@ impl d::Device<B> for Device {
     {
         // Pipeline layouts are implemented as RootSignature for D3D12.
         //
+        // Push Constants are implemented as root constants.
```

Comment:
It would be great to have this architectural documentation more visible. For example, [in Metal](https://github.com/gfx-rs/gfx/blob/master/src/backend/metal/src/lib.rs#L1-L52) I tried to put everything in the doc comments of the crate, so that it's visible in the auto-generated documentation.